packages/ecs/src/ComponentFunctions.ts

```diff
@@ -561,61 +563,78 @@ export const setComponent = <C extends Component>(
+    //   Number(layer),
+    //   val as Entity
+    // ])) {
+    //   if (LayerRelations[entityLayer][linkedLayer] === LayerRelationTypes.Propagate) {
+    // if (!entityExists(linkedEntity)) {
+    //if the linked entity doesn't exist, we recreate it and sync its state with the source entity
+    // linkedEntity = createEntity(dstLayerID as LayerID)
+    // EntityLayerState.linkEntity(entity, linkedEntity)
+    // for (const entityComponent of getAllComponents(entity)) {
+    //   if (entityComponent === component) continue //we're about to do this operation anyways
+    //   setComponent(linkedEntity, entityComponent, entityComponent.stateMap[entity]!.get(NO_PROXY_STEALTH))
+    // }
+    // }
+
+    console.log('cloning component', component.name)
+
+    const argsClone = structuredClone(component.toJSON(getComponent(entity, component))) as SetComponentType<C>
+
+    if (argsClone && component.schema) {
+      //switch any target entities in the args to their corresponding linked entity in the destination layer
+
+      const componentSchema = component.schema as TTypedSchema<C>
+      const frontier = [{ schema: componentSchema, path: '' }] as { schema: any; path: string }[]
+      while (frontier.length > 0) {
+        const { schema, path } = frontier.pop()!
+        const argsClonePath = getNestedObject(argsClone, path).result
+        if (!schema || typeof argsClonePath !== 'object') continue
+        for (const key in argsClonePath) {
+          const valSchema = schema.properties?.[key] as any
+          //check if the value is an entity
+          if (
+            valSchema?.properties?.options &&
+            valSchema.properties.options['id'] === 'Entity'
+            //  && setArgs[key] !== UndefinedEntity
+          ) {
+            const respectiveComponent = getOptionalComponent(linkedEntity, component)
+            if (!respectiveComponent) continue
+            const val = getNestedObject(respectiveComponent, path + '.' + key).result
+            //if so, we need to switch it to the linked entity in the destination layer
+            argsClonePath[key] = val
+          } else if (typeof argsClonePath[key] === 'object') {
+            frontier.push({ schema: valSchema, path: path + '.' + key })
           }
         }
-
-        //set up reactive logic to propagate component changes to linked entity
-        // console.log(dstEntity)
-        setComponent(dstEntity, component, args)
       }
     }
+
+    console.log(argsClone)
+
+    //set up reactive logic to propagate component changes to linked entity
+    // console.log(dstEntity)
+    setComponent(linkedEntity, component, argsClone)
+    // }
+    // }
   }
 
-  propagate()
+  // propagate()
 
-  if (!componentExists && !component.reactorMap.has(entity)) {
+  if (!componentExists && !component.reactorMap.has(entity) && hasComponent(entity, SimulationLayerComponent)) {
```

Comment:
Can't disable creating reactors for all entities that are not simulation layer. We want to be able to have reactive logic in other layers too

Reply:
I disagree with this.

Reply:
resolved